### ISSUE TITLE
494. Target Sum test cases missing and Golang runtime bug

### ISSUE THREAD

**Issue by nevet** [LeetCode-Feedback/LeetCode-Feedback#416]
#### Your LeetCode username
nevetshd

#### Category of the bug
- [ ] Question
- [x] Solution
- [ ] Language

#### Description of the bug
TL;DR: there are 2 bugs in this question:
1. overflow case is missing, and standard code can't run for overflow case;
2. Golang runtime probably has some bugs, the output produced by code (run the same test case in the console) is not the same as the one output during submission.

**Bug 1**

There are only 3 constraints in the statement:

- The length of the given array is positive and will not exceed 20.
- The sum of elements in the given array will not exceed 1000.
- Your output answer is guaranteed to be fitted in a 32-bit integer.

It's possible that intermediate result can't be fitted in a 32-bit integer, consider this test case:
```
[2147483647, 1, 1]
2147483647
```

This case satisfies all constraints, but the standard code can't even be run, it produces the following error:
```
Line 118: Char 18: runtime error: signed integer overflow: 2147483647 + 1 cannot be represented in type 'int' (stl_numeric.h)
```

**Bug 2**

Also, I doubt there are some problems with Golang runtime, my output is the same as the expected output by running in the console, but it always shows differently when submitting the code.

#### Code you used for Submit/Run operation

```
type state struct {
    cur int
    S int64
}

var memo = map[state]int{}

func dfs(nums []int, cur int, S int64) int {
    if cur == len(nums) {
        if S == 0 {
            return 1
        }
        
        return 0
    }
    
    if val, ok := memo[state{cur: cur, S: S}]; ok {
        return val
    }
    
    state := state{cur: cur, S: S}
    memo[state] = dfs(nums, cur + 1, S - int64(nums[cur])) + dfs(nums, cur + 1, S + int64(nums[cur]))
    
    // fmt.Println(state, memo[state])
    
    return memo[state]
}

func findTargetSumWays(nums []int, S int) int {
    return dfs(nums, 0, int64(S))
}
```

#### Language used for code
Golang

#### Expected behavior

The output should be the same when running in console and running in the submission.

#### Screenshots

![image](https://user-images.githubusercontent.com/6215274/83920134-ae619400-a7ae-11ea-91e6-a41a827cee86.png)

We see the same case produced different answers when doing submission and trail run.

**Comment by nevet** [comment]
for the Golang runtime problem, i found it's a common case: https://github.com/LeetCode-Feedback/LeetCode-Feedback/issues/418

**Comment by CooperCooperCooper** [comment]
Hi @nevet,
Thank you for reaching out to us. For the bug 2, the reasons and solutions for different "outputs" from "Run Code" and "Submit" are well documented in [this support article](https://support.leetcode.com/hc/en-us/articles/360011834174-I-encountered-Wrong-Answer-Runtime-Error-for-a-specific-test-case-When-I-test-my-code-using-this-test-case-it-produced-the-correct-output-Why-). Please check it out.

I've relayed the Golang runtime issue to our team to investigate.

**Comment by CooperCooperCooper** [comment]
Hi @nevet ,
Thank you for your time. After a thorough review of the test case contribution, we found it to be violating constraint number 2, "The sum of elements in the given array will not exceed 1000".
